Neos.Flow/Classes/Cli/ConsoleOutput.php

```diff
@@ -239,85 +295,127 @@ public function askHiddenResponseAndValidate($question, $validator, $attempts =
+        $this->getProgressBar()->start($max);
     }
 
     /**
      * Advances the progress output X steps
      *
      * @param integer $step Number of steps to advance
-     * @param Boolean $redraw Whether to redraw or not
      * @return void
      * @throws \LogicException
      */
-    public function progressAdvance($step = 1, $redraw = false)
+    public function progressAdvance(int $step = 1): void
     {
-        $this->getProgressHelper()->advance($step, $redraw);
+        $this->getProgressBar()->advance($step);
     }
 
     /**
      * Sets the current progress
      *
      * @param integer $current The current progress
-     * @param Boolean $redraw Whether to redraw or not
      * @return void
      * @throws \LogicException
      */
-    public function progressSet($current, $redraw = false)
+    public function progressSet(int $current): void
     {
-        $this->getProgressHelper()->setCurrent($current, $redraw);
+        $this->getProgressBar()->setProgress($current);
     }
 
     /**
      * Finishes the progress output
      *
      * @return void
      */
-    public function progressFinish()
+    public function progressFinish(): void
+    {
+        $this->getProgressBar()->finish();
+    }
+
+    /**
+     * @param OutputInterface $output
+     */
+    public function setOutput(OutputInterface $output): void
+    {
+        $this->output = $output;
+    }
+
+    /**
+     * @return OutputInterface
+     */
+    public function getOutput(): OutputInterface
     {
-        $this->getProgressHelper()->finish();
+        return $this->output;
     }
 
     /**
-     * Returns or initializes the symfony/console DialogHelper
+     * @param InputInterface $input
+     */
+    public function setInput(InputInterface $input): void
+    {
+        $this->input = $input;
+    }
+
+
+    /**
+     * @return InputInterface
+     */
+    public function getInput():InputInterface
+    {
+        return $this->input;
+    }
+
+    /**
+     * Returns or initializes the symfony/console QuestionHelper
      *
-     * @return DialogHelper
+     * @return QuestionHelper
      */
-    protected function getDialogHelper()
+    protected function getQuestionHelper(): QuestionHelper
     {
-        if ($this->dialogHelper === null) {
-            $this->dialogHelper = new DialogHelper();
+        if ($this->questionHelper === null) {
+            $this->questionHelper = new QuestionHelper();
             $helperSet = new HelperSet([new FormatterHelper()]);
-            $this->dialogHelper->setHelperSet($helperSet);
+            $this->questionHelper->setHelperSet($helperSet);
         }
-        return $this->dialogHelper;
+        return $this->questionHelper;
+    }
+
+    /**
+     * If question is an array, split it into multi-line string
+     *
+     * @param array $question
+     * @return string
+     */
+    protected function splitQuestion(array $question): string
```

Comment:
nitpick: The naming is strange - a split method that implodes. Would be more a `combineQuestion`?

Reply:
Fully agree.

Reply:
yea, it's about splitting the array into multiple lines, but i see the confusion.  Will go with `combineQuestion` instead

Reply:
I'd go even further and embed the `is_array` into the method as well, then unconditionally call it where it's used. This way it seems half-baked, not only because the docblock says "If question is an array"… Comes with the price of not being able to use the `array` type hint, but…